### DISCUSSION TITLE
Q: Have Pyfunceble set CNAME's inactive (and catch target)

### DISCUSSION THREAD

**Discussion by spirillen** [funilrys/PyFunceble#88] [Q&A] [answered]
## Intro
As I was going to work on this issue <https://www.mypdns.org/T666> It make me think.... 

As it appears, I haven't found any writing on it, that g have started to redirect all of `blogspot.!com` country specific domains to `.com`.

The thought is, that if a record is a `CNAME` then the given record is not needed in e.g. hosts file but the target point is... the CNAME record is only keeping the number of records up slowing down everything and demand further resource's. And some OS simply fails loading to huge hosts files.

## Question
1. Is it currently possible to generate a test line that can at least set CNAME records INACTIVE/INVALID it is **NOT** a A/AAAA/SRV record
1. Are we talking a feature request to have Pyfunceble fetching CNAME set them inactive and return CNAME target?
1. If 2, then this would output 2 new files
    1. output/type/redirect/list
    1. output/type/redirect/target

## General
Please all @pyfunceble users contribute your thought on this in general, would this be a good idea or not?

How are you other dealing with this?

**Comment by funilrys** [comment, marked as answer]
Hi @spirillen,

PyFunceble will try to fetch the `NS`, `CNAME`, `DNAME`, `A` and `AAAA` respectively before flagging a subject as `INACTIVE`. And, as soon as it meets a response from one of them (NS, CNAME, ...) It stops and declares the subject as `ACTIVE`. Otherwise, if they were not all successful, it returns it as `INACTIVE`.

Now, you are talking about the Blogspot sub-domains. They are part of our SPECIAL rules (if activated): https://pyfunceble.readthedocs.io/en/latest/responses/index.html#blogspot that means that PyFunceble will still follow the redirect in order to determine if it is applicable to the SPECIAL rules.


-----------------------------------
> Is it currently possible to generate a test line that can at least set CNAME records INACTIVE/INVALID it is NOT a A/AAAA/SRV record

Actually not in my plan but we can look into that on a deeper level. If we add something like that, it will still stay an option. 

What I can do immediately is to add a CNAME category to the analytic directory. That might be the easier part right now. 

Other questions can be answered along with the implementation of such an option. As said, I'm more open to add a `CNAME` category/file into the analytic directory, but if an `NS` record comes before, the system will never get the `CNAME` because we don't actually request it in that case.

My only concern with:
>  fetching CNAME set them inactive and return CNAME target

It can become really quickly a never-ending test for the case that a system meets a CNAME of a CNAME of a CNAME and so on... I know it's not common but in the wild, it might be more common than we think...


Stay safe and healthy!
Nissar

**Comment by spirillen** [comment]
There it was... the bs rule, thought I've seen it somewhere :smiley: 

All `301`, `302 and `404` returns `INACTIVE`

So the best question here might be:

Does anyone actually knows anything about this domain ? rather than guessing 

Again, what are you other doing with these CNAME fillers in your sources? And could it be relevant to have @funilrys working on this for you?

UPDATE: rereaded the docs :stuck_out_tongue_closed_eyes:
